### DISCUSSION TITLE
How to enter Configuration Mode

### DISCUSSION THREAD

**Discussion by EseJota11** [Mustard2/MustardUI#82] [Help] [answered]
Hello, I come here to ask for help, in something that I am almost sure the solution is something very simple, but I just can't figure it out
It is the first time that I use the UI, it guided me step by step in the configuration guide and everything went well, but where I got stuck is with the custom properties, and the options are not displayed when I right click on a property , except if I have selected the rig, but if for example I want to assign a shape key, therefore I select the object in question, I go to the shape key and when I right click the options to add them as custom property do not appear
I see that in the configuration guide it talks about going to "Configuration mode" before right-clicking on a property, and I suppose the solution is there, the detail is that I don't know what exactly is meant by ""Configuration mode ", or the context of this, how it is activated (if it is the case) or what to do exactly
Thank you very much in advance for your attention, sorry for the inconvenience and have a good day

**Comment by Mustard2** [comment, marked as answer]
### NOTE: this is deprecated. To enter Configuration mode, follow https://github.com/Mustard2/MustardUI/wiki/Configuration

Hi!

As you were saying you can add custom properties only in Configuration mode. I made this choice to avoid unaware users to accidentally add or delete custom properties.

In order to enter Configuration mode, you really just need to switch to the configuration panel, and this can be done going to the **Settings** panel, then clicking on the **Maintenance mode button** and finally clicking on the **Configure UI** button, as in the image attached. Again, this procedure is a bit long on purpose, to avoid users of your model to make accidental changes to the UI.

![Immagine 2022-11-18 000536](https://user-images.githubusercontent.com/60780948/202578994-a47312ca-6c31-4ad7-87b2-810c4fa644ae.png)

If you are in Configuration mode, you should see something like this. Then you should be able to add custom properties right clicking on Blender properties.

![Immagine 2022-11-18 000758](https://user-images.githubusercontent.com/60780948/202579165-43f6efa0-7012-488b-acbf-8f41cdb5bdb3.png)

Please let me know if you still have issues (feel free to open other questions if you need more help on other stuffs :) ).

**Reply by EseJota11** [reply]
If it's solved, thanks a million! 🙏
That would be all for now, thanks again and have a nice day
